File: utils/is_line_equal_test.go
package utils

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestIsLineEqual_EqualCase(t *testing.T) {
	line := []string{"1", "2", "3"}

	assert.True(t, IsLineEqual(line, line))
}

func TestIsEqual_LengthMismatch(t *testing.T) {
	line1 := []string{"1", "2", "3"}
	line2 := []string{"1", "2", "3", "4"}

	assert.False(t, IsLineEqual(line1, line2))
}

func TestIsEqual_UnequalCase(t *testing.T) {
	line1 := []string{"1", "3", "2"}
	line2 := []string{"1", "2", "3"}

	assert.False(t, IsLineEqual(line1, line2))
}

File: utils/isLineEqual.test.ts
import { expect } from "@std/expect";
import { describe, it } from "@std/testing/bdd";
import { isLineEqual } from "./isLineEqual.ts";

describe("isLineEqual", () => {
  it("Equal case", () => {
    const line = ["1", "2", "3"];
    expect(isLineEqual(line, line)).toBe(true);
  });

  it("Length mismatch", () => {
    const line1 = ["1", "2", "3"];
    const line2 = ["1", "2", "3", "4"];
    expect(isLineEqual(line1, line2)).toBe(false);
  });

  it("Unequal case", () => {
    const line1 = ["1", "3", "2"];
    const line2 = ["1", "2", "3"];
    expect(isLineEqual(line1, line2)).toBe(false);
  });
});
